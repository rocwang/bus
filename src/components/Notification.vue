<template>
  <transition
    :enter-class="$style.enterAndLeaveTo"
    :leave-to-class="$style.enterAndLeaveTo"
    :enter-active-class="$style.active"
    :leave-active-class="$style.active"
  >
    <div :class="$style.root" @keyup.esc="$emit('close')">
      <slot />
    </div>
  </transition>
</template>

<script>
export default {
  name: "Notification"
};
</script>

<style module>
.root {
  position: fixed;
  z-index: 9999;
  top: 0;
  right: 0;
  left: 0;
  height: auto;
  padding: 15px;
  color: var(--c-white);
  background-color: var(--c-ocean-blue);
  box-shadow: 2px 2px 4px 0 var(--c-ocean-blue);
}

.active {
  transition: transform 0.3s ease-out;
}

.enterAndLeaveTo {
  transform: translateY(-100%);
}
</style>
