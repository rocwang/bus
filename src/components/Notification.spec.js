import { mount } from "@vue/test-utils";
import Notification from "./Notification";

describe("Notification.vue", () => {
  it("matches the snapshot", () => {
    const wrapper = mount(Notification);
    expect(wrapper.element).toMatchSnapshot();
  });

  it("emits a 'close' event when press ESC", () => {
    const wrapper = mount(Notification);
    wrapper.trigger("keyup.esc");
    expect(wrapper.emitted("close")).toBeTruthy();
  });

  it("has a default slot", () => {
    const wrapper = mount(Notification, {
      slots: {
        default: "<button/>"
      }
    });
    expect(wrapper.contains("button")).toBe(true);
  });
});
