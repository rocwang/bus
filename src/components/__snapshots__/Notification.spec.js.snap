// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification.vue matches the snapshot 1`] = `
<div
  class="root"
  enter-active-class="active"
  enter-class="enterAndLeaveTo"
  leave-active-class="active"
  leave-to-class="enterAndLeaveTo"
/>
`;
